lib/Tomcat/Utils.pm

```diff
@@ -68,20 +71,28 @@ EOF
 }
 
 
-# Access the tomcat web application manager
+# Check Servlet, JSP and Websocket, the example files can be accessed, login with admin via tomcat manager
 sub tomcat_manager_test() {
-    my ($self) = shift;
-
-    $self->firefox_open_url('localhost:8080/manager');
-    assert_screen('tomcat-manager-authentication', 180);
-    type_string('admin');
-    send_key('tab');
-    type_string('admin');
-    assert_and_click('tomcat-OK-autentication');
-    wait_still_screen(2);
-    assert_and_click('tomcat-click-save-login');
-    assert_screen('tomcat-web-application-manager', TIMEOUT);
+    select_serial_terminal;
+
+    # sometimes we have network issue (cannot connect to localhost)
+    systemctl('restart network.service');
```

Comment:
instead of hitting this with a big hammer, I'd rather want to know when this does not work (and why).

Maybe: first check if there is an issue to connect to localhost (should hardly ever be an issue to ping ::1 or 127.0.0.1); in case of failure, record soft_failure, restart network, retest and fail on error

Reply:
sporadic issue, really hard to determine, I am open for suggestion though.